<launch>
  <!-- General arguments -->
  <arg name="debug_mode"
       default="false"
       doc="Run the mapper in debug mode." />
  <arg name="show_rviz"
       default="true"
       doc="Automatically launch Rviz." />

  <!-- Planner arguments -->
  <arg name="planner_config" default="glocal_planner.yaml"/>
  
  <!-- Airsim arguments -->
  <arg name="airsim_config" default="maze_airsim.yaml"/>
  <arg name="use_airsim_time" default="true"/>
  <arg name="mav_name" default="airsim_drone"/>
  
  <!-- Experiment arguments -->
  <arg name="record_data" default="true"/>
  <arg name="data_path" default="/home/lukas/Documents/Glocal/Data"/>
  
  <!-- Automatically generated arguments -->
  <arg unless="$(arg debug_mode)" name="launch_prefix" value="" />
  <arg if="$(arg debug_mode)" name="launch_prefix" value="gdb -ex run --args" />




  
  <!-- Setup the experiment -->
  <param name="/use_sim_time" value="true" if="$(arg use_airsim_time)"/>
  
  <!-- tf -->
  <node pkg="tf" type="static_transform_publisher" name="static_tf_odom_world" args="0 0 0 0 0 0 1 /odom /world 100"/>
  <node pkg="tf" type="static_transform_publisher" name="static_tf_mission_world" args="0 0 0 0 0 0 1 /world /mission 100"/>
  
  <!-- airsim client -->
  <node name="airsim_simulator" pkg="unreal_airsim" type="airsim_simulator_node" required="true" output="screen" args="-alsologtostderr">
     <rosparam file="$(find glocal_exploration_ros)/config/$(arg airsim_config)"/>
  </node>

  <!-- Planner -->
  <group ns="glocal">
    <node name="glocal_system" pkg="glocal_exploration_ros" type="glocal_system_node"
          output="screen" launch-prefix="$(arg launch_prefix)">
      <rosparam command="load" file="$(find glocal_exploration_ros)/config/$(arg planner_config)"/>
      <remap from="~pointcloud" to="/$(arg mav_name)/Lidar" />
      <remap from="odometry" to="/$(arg mav_name)/ground_truth/odometry" />
      <remap from="command/pose" to="/$(arg mav_name)/command/pose" />
      <remap from="~/global_planner/submap_in" to="~/mapping/submap_esdfs" />
      <remap from="~/global_planner/submap_pose_in" to="~/mapping/submap_poses" />
    </node>  
  </group>
  
  <!-- Experiment startup and Evaluation -->
  <group ns="evaluation">
    <node name="experiment_manager" pkg="glocal_exploration_ros" type="experiment_manager.py" output="screen">
  
    </node>
    <node name="voxblox_gt" pkg="voxblox_ros" type="esdf_server" if="$(arg record_data)">
      <rosparam command="load" file="$(find glocal_exploration_ros)/config/voxblox_ground_truth.yaml"/>
      <remap from="pointcloud" to="/$(arg mav_name)/Lidar"/>
    </node>
  </group>
  
  <!-- Visualization -->
  <group if="$(arg show_rviz)">
    <arg name="rviz_file" default="$(find glocal_exploration_ros)/config/maze_visualization.rviz" />
   <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rviz_file)"/>
  </group>
</launch>
